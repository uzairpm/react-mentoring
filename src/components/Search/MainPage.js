import React, {Component} from 'react';

import { Route, Link } from "react-router-dom";

import Constants from '../Common/constants';
import Header from '../Header/Header';
import SearchBox from '../Search/SearchBox';
import SearchResults from '../Search/SearchResults';

export default class MainPage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            movies: [],
            titleActive: true,
            sortByReleaseDate: true,
            searchText: ''
        };
        this.titleClick = this.titleClick.bind(this);
        this.genreClick = this.genreClick.bind(this);
        this.searchClick = this.searchClick.bind(this);
        this.releaseDateClick = this.releaseDateClick.bind(this);
        this.ratingClick = this.ratingClick.bind(this);
        this.valueChangeHandler = this.valueChangeHandler.bind(this);
    }
    refreshSearchResults() {
        let url = Constants.baseURL + '?search='+this.state.searchText+'&searchBy=';
        if(this.state.titleActive === true) {
            url += 'title';
        } else {
            url += 'genres';
        }
        if (this.state.sortByReleaseDate === true) {
            url += '&sortBy=release_date&sortOrder=desc';
        } else {
            url += '&sortBy=vote_average&sortOrder=desc';
        }
        fetch(url)
            .then(res => res.json())
            .then(response => {
                this.setState({
                    movies: response.data
                });
            });
    }
    searchClick() {
        this.refreshSearchResults();
    }
    titleClick() {
        this.setState({ titleActive: true });
    }
    genreClick() {
        this.setState({ titleActive: false });
    }
    releaseDateClick() {
        this.setState({ sortByReleaseDate: true }, () => {
            this.refreshSearchResults();
        });
    }
    ratingClick() {
        this.setState({ sortByReleaseDate: false }, () => {
            this.refreshSearchResults();
        });
    }
    valueChangeHandler(e) {
        this.setState({
            searchText: e.target.value
        });
    }
    render() {
        return (
            <div>
                <Header showSearchButton={false} />
                <SearchBox titleActive={this.state.titleActive}
                    value={this.state.searchText}
                    valChange={this.valueChangeHandler}
                    titleClick={this.titleClick}
                    genreClick={this.genreClick}
                    searchClick={this.searchClick}/>
                <SearchResults movies={this.state.movies}
                    releaseDate={this.state.sortByReleaseDate}
                    releaseDateClick={this.releaseDateClick}
                    ratingClick={this.ratingClick} />
            </div>
        );
    }
}
