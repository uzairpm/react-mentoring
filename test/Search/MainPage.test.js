import React from 'react';

import MainPage from '../../src/components/Search/MainPage';

describe('Check Component creation', () => {
    const element = shallow(<MainPage />);
    test('MainPage component', () => {
        expect(element).toMatchSnapshot();
    });
    test('MainPage state', () => {
        element.instance().handleTitleClick();
        expect(element.state('titleActive')).toEqual(true);
    });
    test('MainPage state 2', () => {
        element.instance().handleGenreClick();
        expect(element.state('titleActive')).toEqual(false);
    });
    test('MainPage state 3', () => {
        element.instance().valueChangeHandler({target: {value: 'mat'}});
        expect(element.state('searchText')).toEqual('mat');
    });
    /* test('MainPage state 4', () => {
        element.instance().handleReleaseDateClick();
        expect(element.state('sortByReleaseDate')).toEqual(true);
    });
    test('MainPage state 5', () => {
        element.instance().handleRatingClick();
        expect(element.state('sortByReleaseDate')).toEqual(false);
    }); */
});
