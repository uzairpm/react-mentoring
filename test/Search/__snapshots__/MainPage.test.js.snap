// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component Snapshot test with default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          showSearchButton={false}
        />,
        <Connect(SearchResults) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "showSearchButton": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            showSearchButton={false}
          />,
          <Connect(SearchResults) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "showSearchButton": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
